models/user.py

```diff
@@ -68,7 +76,7 @@ def hash_pw(plaintext_password):
         return bcrypt.hashpw(bytes(plaintext_password, 'utf-8'), bcrypt.gensalt())
 
     def check_pw(self, plaintext_password):
-        return bcrypt.checkpw(bytes(plaintext_password, 'utf-8'), self.password)
+        return bcrypt.checkpw(bytes(plaintext_password, 'utf-8'), self._password)
```

Comment:
small detail: I think we should keep this as `self.password` which does end up returning `self._password`, however it is best practice to use getter methods for accessing private fields.